File: WolfCafe/wolf-cafe-backend/src/main/java/edu/ncsu/csc326/wolfcafe/service/OrderService.java
package edu.ncsu.csc326.wolfcafe.service;

import java.util.List;

import edu.ncsu.csc326.wolfcafe.dto.ItemDto;
import edu.ncsu.csc326.wolfcafe.dto.OrderDto;

/**
 * Order service
 *
 * @author Shakthi Ravichandran
 */
public interface OrderService {

    /**
     * String for ordered state.
     */
    public final String ORDER_ORDERED    = "Ordered";

    /**
     * String for fulfilling state.
     */
    public final String ORDER_FULFILLING = "Fulfilling";

    /**
     * String for fulfilled state.
     */
    public final String ORDER_FULFILLED  = "Fulfilled";

    /**
     * String for picked up state.
     */
    public final String ORDER_PICKED_UP  = "Picked up";

    /**
     * String for cancelled state.
     */
    public final String ORDER_CANCELLED  = "Cancelled";

    /**
     * Adds items to order.
     *
     * @param orderId
     *            ID of order to add to.
     * @param orderItemDtos
     *            items to add.
     * @return updated OrderDto.
     */
    public OrderDto addItemToOrder ( Long orderId, List<ItemDto> orderItemDtos );

    /**
     * Adds tip to order.
     *
     * @param orderId
     *            ID of order to add tip to.
     * @param tipPercent
     *            percentage of tip.
     * @return updated OrderDto.
     */
    public OrderDto addTipToOrder ( Long orderId, Double tipPercent );

    /**
     * Gets total price of order.
     *
     * @param order
     *            order to get price of.
     * @return double price of order.
     */
    public Double calculateTotalPrice ( OrderDto order );

    /**
     * Gets order using order ID.
     *
     * @param orderId
     *            ID of order.
     * @return relevant OrderDto.
     */
    public OrderDto getOrderById ( Long orderId );

    /**
     * Creates a new order.
     *
     * @param orderDto
     *            data to populate new order with.
     * @return new OrderDto.
     */
    public OrderDto createOrder ( OrderDto orderDto );

    /**
     * Updates order status with the passed string.
     *
     * @param status
     *            the status.
     * @param orderDto
     *            the order to update.
     * @return new OrderDto.
     */
    public OrderDto updateOrder ( String status, OrderDto orderDto );

    /**
     * Returns all orders
     *
     * @return all orders
     */
    List<OrderDto> getAllOrders ();
}

File: WolfCafe/wolf-cafe-frontend/src/services/OrderService.js
import axios from 'axios'


const BASE_REST_API_URL = 'http://localhost:8080/api/order'

export const ORDER_STATUS = {
    ORDERED: "Ordered",
    FULFILLING: "Fulfilling",
    FULFILLED: "Fulfilled",
    PICKED_UP: "Picked up",
    CANCELLED: "Cancelled"
}

/** GET Orders - gets all orders */
export const getAllOrders = () => {
    return axios.get(BASE_REST_API_URL);
}

/** GET Order - gets the Order with the given id */
export const getOrderById = (id) => {
    return axios.get(`${BASE_REST_API_URL}/${id}`);
}

/** POST Order - creates the given Order */
export const createOrder = (orderDto) => {
    return axios.post(BASE_REST_API_URL, orderDto);
}

/** PUT Order - updates the order with the given id */
export const updateOrder = (id, status) => {
    console.log('Sending status update for order:', id, 'new status:', status);
    return axios.post(`${BASE_REST_API_URL}/${id}/status`, 
        JSON.stringify(status),
        {
            headers: {
                'Content-Type': 'application/json'
            }
        }
    );
}

